### PR TITLE
Update driver_load and add more scenerio

### PR DESCRIPTION
**Pull request by suqinhuang**
Update driver_load and add more scenerio

Signed-off-by: Suqin Huang

### CODE REVIEW

File: qemu/tests/driver_load.py [hunk 1]
```diff
@@ -36,26 +68,25 @@ def run(test, params, env):
...
-
-    driver_id = driver_id[0]
-    if params["os_type"] == "windows":
-        driver_id = '^&'.join(driver_id.split('&'))
-    driver_load_cmd = driver_load_cmd.replace("DRIVER_ID", driver_id)
-    driver_unload_cmd = driver_unload_cmd.replace("DRIVER_ID", driver_id)
+    error.context("Copy devcon.exe from winutils.iso to C:\\")
+    copy_devcon_cmd = params.get("devcon") % \
+                      utils_misc.get_winutils_vol(session)
+    session.cmd(copy_devcon_cmd)
```
Comment (spcui) [inline qemu/tests/driver_load.py]: for this step, should be only windows.

File: qemu/tests/cfg/driver_load.cfg [hunk 2]
```diff
@@ -2,35 +2,70 @@
     type = driver_load
     kill_vm_on_error = yes
     login_timeout = 240
-    driver_load_cmd = modprobe DRIVER_ID
-    driver_unload_cmd = modprobe -r DRIVER_ID
     repeats = 1
     variants:
-        - nic_device:
+        - with_nic:
             no e1000
```
Comment (spcui) [inline qemu/tests/cfg/driver_load.cfg]: why not support e1000?

File: qemu/tests/cfg/driver_load.cfg [hunk 3]
```diff
@@ -2,35 +2,70 @@
...
-            Windows:
-                # The DevCon utility is a command-line utility that acts as an
-                # alternative to Device Manager. Please set up your windows guest
-                # following: http://support.microsoft.com/kb/311272/en-us
-                driver_id_cmd = C:\devcon.exe find * | find "VirtIO Ethernet Adapter"
-                driver_id_pattern = "(.*?):"
-                driver_load_cmd = C:\devcon.exe enable DRIVER_ID
-                driver_unload_cmd = C:\devcon.exe disable DRIVER_ID
-        - balloon_device:
+            driver_unload_cmd = modprobe - r virtio_net
```
Comment (spcui) [inline qemu/tests/cfg/driver_load.cfg]: typo error? should be "-r"

File: qemu/tests/cfg/driver_load.cfg [hunk 4]
```diff
@@ -2,35 +2,70 @@
...
             balloon_dev_add_bus = yes
-            driver_id_cmd = lsmod |grep virtio_balloon
+            driver_id_cmd = lsmod | grep virtio_balloon
             driver_id_pattern = virtio_balloon
-            Windows:
-                driver_id_cmd = C:\devcon.exe find * | find "VirtIO Balloon Driver"
-                driver_id_pattern = "(.*?):"
-                driver_load_cmd = C:\devcon.exe enable DRIVER_ID
-                driver_unload_cmd = C:\devcon.exe disable DRIVER_ID
+            driver_unload_cmd = modprobe - r virtio_balloon
```
Comment (spcui) [inline qemu/tests/cfg/driver_load.cfg]: and this line

File: qemu/tests/cfg/driver_load.cfg [hunk 5]
```diff
@@ -2,35 +2,70 @@
...
+            driver_load_cmd = modprobe virtio_balloon
+
+        - with_viorng:
+            virtio_rngs += " rng0"
+            backend_rng0 = rng - random
+            backend_type = passthrough
+            filename_passthrough = /dev / random
+            driver_id_cmd = lsmod | grep virtio_rng
+            driver_id_pattern = virtio_rng
+            driver_unload_cmd = modprobe - r virtio_rng
```
Comment (spcui) [inline qemu/tests/cfg/driver_load.cfg]: should be "modprobe -r"

**Comment by suqinhuang** [comment]
@spcui update, could you review it again

File: qemu/tests/cfg/driver_load.cfg (hunk 2 above)
Comment (suqinhuang) [inline qemu/tests/cfg/driver_load.cfg]: this case is virtio only, rtl8139 is used as nic2, so, only no e1000

File: qemu/tests/cfg/driver_load.cfg [hunk 6]
```diff
@@ -2,35 +2,70 @@
...
-                driver_id_cmd = C:\devcon.exe find * | find "VirtIO Balloon Driver"
-                driver_id_pattern = "(.*?):"
-                driver_load_cmd = C:\devcon.exe enable DRIVER_ID
-                driver_unload_cmd = C:\devcon.exe disable DRIVER_ID
+            driver_unload_cmd = modprobe -r virtio_balloon
+            driver_load_cmd = modprobe virtio_balloon
+
+        - with_viorng:
+            virtio_rngs += " rng0"
+            backend_rng0 = rng - random
```
Comment (spcui) [inline qemu/tests/cfg/driver_load.cfg]: should be "rng-random"

File: qemu/tests/cfg/driver_load.cfg [hunk 7]
```diff
@@ -2,35 +2,70 @@
...
+            backend_type = passthrough
+            filename_passthrough = /dev / random
```
Comment (spcui) [inline qemu/tests/cfg/driver_load.cfg]: /dev /random

Comment (suqinhuang) [inline qemu/tests/cfg/driver_load.cfg]: ok, something strange, thanks for review

File: qemu/tests/driver_load.py [hunk 8]
```diff
@@ -22,6 +21,41 @@ def run(test, params, env):
...
+                raise error.TestFail("failed to unload driver, %s" % output)
+
+    def check_driver(session, cmd, pattern):
+        output = session.cmd_output(cmd)
+        driver_id = re.findall(pattern, output)
+        if not driver_id:
+            raise error.TestFail("Didn't find driver info from guest %s"
+                                 % output)
+
+        driver_id = driver_id[0]
```
Comment (xutian) [inline qemu/tests/driver_load.py]: can you add comments why need this step?

**Comment by xutian** [comment]
other parts looks fine.

thanks,
Xu

**Comment by xutian** [comment]
got it thanks.

**Pull request merged**